hedyorg/hedy
# 💻  Use subdomains to navigate through languages

Makes it possible to set the language through subdomains. In order to do this the following changes were made:

* Every route in the page was duplicated to include a `subdomain` argument, otherwise pages wouldn't be accesible without being prefixed with the subdomain.
* The set up language function was edited to take the subdomain as a parameter


Fixes #5720

**How to test**

In order to test this locally, we need to set an env variable that has a dot in it. The one I used was `hedy.local`, this name should be added to the hosts file in your system prefixed with the languages you want to test, for example

```
127.0.0.1 es.hedy.local hedy.local en.hedy.local
```

After doing this, before starting the server, you need to set the `DOMAIN_NAME` env variable `export DOMAIN_NAME=hedy.local`

After doing this you are set up to test the feature. You can navigate to nl.hedy.local:8080 and see that the language changes, and change it through the language selector. Also log in as a user, and change the preferred language there
